### ISSUE TITLE
Writing into EvenHubs, Spark executors fail silently when getting throttled

### ISSUE THREAD

**Issue by Sridhar-MS** [Azure/azure-event-hubs-spark#460]
I have a job running in databricks which writes the result of  a batch query (which returns 7 millions rows) into an eventhub with 1 TU. This job completes successfully but when I inspect the eventhub I see only around 2million events getting published. 

On further inspection of the executor logs I see `com.microsoft.azure.eventhubs.ServerBusyException` being thrown but this error does not abort the spark job. Therefore its possible that the write operation returns without any exception but still fail to publish all the events.

Sample code:
```
val ehWriteConf = EventHubsConf(peersEhConnStr)
val peersLAFrame = peersFrame.select($"body", $"properties")
peersLAFrame.write
               .format("eventhubs")
               .options(ehWriteConf.toMap)
               .save()
```

Bug Report:

- Actual behavior
  Throttled eventhub write/ingress operations fail silently.
- Expected behavior
  When the eventhub operation is throttled that error should possibly abort the job.
- Spark version
   Databricks 5.5 LTS (includes Apache Spark 2.4.3, Scala 2.11)
- spark-eventhubs artifactId and version
   azure-eventhubs-spark_2.11:2.3.13

**Comment by nyaghma** [comment]
Could you please upgrade to the current version (2.3.14.1) and let us know if you see the error again.

**Comment by nyaghma** [comment]
I'm closing this issue now. Please reopen if you see the same problem using the current version.

**Comment by rvoak** [comment]
@nyaghma I was able to reproduce the same issue using the latest version (2.3.15). The spark executors fail silently, and **not only while getting throttled.** 

The write runs successfully irrespective of any exceptions that arise during execution of the write operation. I ran the same code snippet as mentioned above, and faced the same issue. I was also able to reproduce the same issue when I tried to write by passing in an **incorrect connection string**; the executor still returned a success.
